### ISSUE TITLE
Gem crash when asset_host_spec is a Proc

### ISSUE THREAD

**Issue by GeReV** [wpeterson/emoji#30]
I have a certain use-case where my global (development) asset_host is set to a Proc.

In that case, my development server crashes on startup:

```
.../emoji-1.0.4/lib/emoji.rb:31:in `parse_and_validate_asset_host': undefined method `size' for #<Proc:0x007fd44c4439f0> (NoMethodError)
```

**Comment by wpeterson** [comment]
Thanks for reporting this.

Can you describe your use case so I make sure this works appropriately for you?  You're hoping to pass a proc as the asset host value that returns a URL string?  Are you expecting the proc to be evaluated for each asset that is rendered?  Does it take the asset path as a param?

Let me know the details and I'll add support for this.  It's not a common use case but it sounds like a good one to support.

**Comment by GeReV** [comment]
My use case is based on what's described in [this blog post[(http://kevinold.com/2015/02/04/configure-webpack-dev-server-and-react-hot-loader-with-ruby-on-rails.html) for integrating Webpack Dev Server for Rails when using React.

From what I understand, the Proc evaluates with the name of each asset, and possibly return a different host which is used to build the asset's URL. When nothing is returned, default behavior remains.
Essentially an override for specific assets.

This is the example in the article:

```ruby
# environments/development.rb
config.action_controller.asset_host = Proc.new { |source|
  if source =~ /wp_bundle\.js$/i
  "http://localhost:8080"
  end
}
```

**Comment by wpeterson** [comment]
Thanks for confirming, sounds like the same behavior that Rails has.  I'm planning to add this.

**Comment by wpeterson** [comment]
Fixed in recent release 1.0.5:

https://github.com/wpeterson/emoji/blob/master/CHANGELOG.md#105

**Comment by wpeterson** [comment]
Please grab the latest and let me know if it resolves your issue, I think it should.
